users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -47,14 +47,27 @@ def get_datasets(**kwargs):
         returnn_python_exe=RETURNN_EXE,
         **kwargs,
     )
-
-    returnn_datasets = {
-        "train": nn_train_data_inputs["switchboard.train"].get_data_dict()["datasets"]["ogg"],
-        "dev": nn_cv_data_inputs["switchboard.cv"].get_data_dict()["datasets"]["ogg"],
-        "eval_datasets": {
-            "devtrain": nn_devtrain_data_inputs["switchboard.devtrain"].get_data_dict()["datasets"]["ogg"],
-        },
-    }
+    if useMultiProcDataset:
+        returnn_datasets = {
+            "train": {
+                "class": "MultiProcDataset",
+                "dataset": nn_train_data_inputs["switchboard.train"].get_data_dict()["datasets"]["ogg"],
+                "num_workers": 2,
+                "buffer_size": 5,
+            },
+            "dev": nn_cv_data_inputs["switchboard.cv"].get_data_dict()["datasets"]["ogg"],
+            "eval_datasets": {
+                "devtrain": nn_devtrain_data_inputs["switchboard.devtrain"].get_data_dict()["datasets"]["ogg"],
+            },
+        }
+    else:
+        returnn_datasets = {
+            "train": nn_train_data_inputs["switchboard.train"].get_data_dict()["datasets"]["ogg"],
+            "dev": nn_cv_data_inputs["switchboard.cv"].get_data_dict()["datasets"]["ogg"],
+            "eval_datasets": {
+                "devtrain": nn_devtrain_data_inputs["switchboard.devtrain"].get_data_dict()["datasets"]["ogg"],
+            },
+        }
```

Comment:
You can always use the default (`else` branch) to create `returnn_datasets` and only `if use_multi_proc_dataset` then `returnn_datasets["train"] = {"class": ..., "dataset": returnn_datasets.pop("train")` or similar. That would be easier to read.